src/Components/AddItem.jsx

```diff
@@ -1,5 +1,16 @@
 import React from 'react';
+import { useLocation, Redirect } from 'react-router-dom';
 
-export default function AddItem() {
-  return <div>Hi from Add an Item View!</div>;
+export default function AddItem({ token }) {
+  let location = useLocation().pathname;
+
+  return (
+    <div>
+      {token === null && location === '/add-item' ? (
+        <Redirect to="/" />
+      ) : (
+        <div>Hi from Add an Item View!</div>
+      )}
+    </div>
+  );
```

Comment:
Since this component is only displayed on the `/add-item` route, is the check for location needed? Also this is more of a stylistic choice, but you can return the Redirect in an if statement since it isn't a renderable component.
```javascript
if (token === null && location === '/add-item') {
  return <Redirect to="/" />;
}
```

Reply:
Thanks for this suggestion Terry! We ran into a refactoring issue we couldn't figure out because of location, but removing location, which isn't even needed, solved all of that. This let us clean up the List and Add Item components so all routing logic stayed in Routes.jsx